### DISCUSSION TITLE
AOC Day 14 Discussion: grain by grain

### DISCUSSION THREAD

**Discussion by rhdeck** [Virtual-Coffee/virtualcoffee.io#774] [Advent of Code 2022]
Discussion of solutions for [Advent of Code Day 14](https://adventofcode.com/2022/day/14) here.

Don't forget about the VC Advent of Code private leaderboard:

Go to [leaderboard/private](https://adventofcode.com/2022/leaderboard/private)
Use join code 363496-e6f42283

**Comment by rhdeck** [comment]
A fun little simulator. Got me further into multidimensional numpy arrays! [solve](https://github.com/rhdeck/aoc-2022/blob/main/src/14/day_14.ipynb)

**Reply by doleraj** [reply]
Somehow I didn't even think of making the stone lines a separate data structure. Sometimes the visual representation really locks you into a data representation...

**Comment by doleraj** [comment]
Difficulty was all in the parsing on this one. There's almost certainly a cleaner way to handle that in TS than just looping like I do, but I'm not sure what it would be. https://github.com/doleraj/adventofcode/blob/main/aoc2022/src/day14/index.ts

**Reply by tkshill** [reply]
> // I'm assuming the input doesn't screw with me here. Probably a bad plan.

A good plan! One nice thing about AOC is that you can usually assume the input is _good_, in the sense that any weirdness is probably included in the problem statement. I've maybe been burned by this like, once.

**Comment by tkshill** [comment]
My day 14

Other solutions here are interesting, @rhdeck  I never attempted to render the output except to test. My "cavern" is just a set (because sets are great!) @doleraj I also didn't separate the rocks from the sand. The type of matter ended up not... mattering. 🥁 (I'm so so sorry)

Initially solved without any optimization, took 15 minutes, made my computer angry. Now optimized, less angry.

Also testing some new file formatting. Also improved my rigging for solutions. Happy with this.

[solve](https://github.com/tkshill/AdventOfCode2022/blob/main/AOC2022/Day14.fs)
